systems/plants/TimeSteppingRigidBodyManipulator.m

```diff
@@ -605,11 +605,9 @@ function checkDirty(obj)
               obj.LCP_cache.data.fastqp_active_set = [];
 
               % take M_active, z_inactive on reduced LCP
-              obj.LCP_cache.data.M_active = false(nL+nP+(2+mC)*nC,1);
-              obj.LCP_cache.data.M_active(z_inactive_guess) = ...
-                M(z_inactive_guess,z_inactive_guess)*z(z_inactive_guess)+w(z_inactive_guess)<1e-8;
               obj.LCP_cache.data.z_inactive = false(nL+nP+(2+mC)*nC,1);
               obj.LCP_cache.data.z_inactive(z_inactive_guess) = z(z_inactive_guess)>lb(z_inactive_guess)+1e-8;            
+              obj.LCP_cache.data.M_active = obj.LCP_cache.data.z_inactive;
             end
 %             if isempty(active_set_fail_count)
```

Comment:
Sorry, should have put that with the commit. Previously we could have:
1) M_active and not z_inactive. This unnecessarily constrains M_active*z_inactive+w_active=0 at those indices, possibly creating extra QP failures. 
2) z_inactive but not M_active. This is obviously incorrect, but would come about due to tolerance violations in our M_active logic.

Rather than playing numerical games, the simplest thing to me was the logic I proposed. Empirically, it leads to fewer active-set failures (on atlas falling example) and fixes the cases when (2) would cause simulation explosions.